pkg/network/http/shared_libraries_test.go

```diff
@@ -0,0 +1,216 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package http
+
+import (
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"sync"
+	"testing"
+	"time"
+
+	"go.uber.org/atomic"
+
+	ddebpf "github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	netebpf "github.com/DataDog/datadog-agent/pkg/network/ebpf"
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
+)
+
+func TestSharedLibraryDetection(t *testing.T) {
+	skipTestIfKernelNotSupported(t)
+	if !httpsSupported() {
+		t.Skip("shared library detection not supported. skipping.")
+	}
+
+	perfHandler, doneFn := initEBPFProgram(t)
+	fpath := filepath.Join(os.TempDir(), "foo.so")
```

Comment:
nit: use `t.TempDir()` and you don't need to worry about removing the file afterwards.

Reply:
nice!